modules/atf_logger.lua

```diff
@@ -23,24 +46,91 @@ function Logger:SDLtoMOB(message)
+  end
 end
 
 function Logger:HMItoSDL(message)
   local log_str = string.format(Logger.hmi_log_format, "HMI->SDL", formated_time(), message)
   self.atf_log_file:write(log_str)
+  if config.storeFullATFLogs then
+    self.full_atf_log_file:write(log_str)
+  end
 end
 
 function Logger:SDLtoHMI(message)
   local log_str = string.format(Logger.hmi_log_format, "SDL->HMI", formated_time(), message)
   self.atf_log_file:write(log_str)
+  if config.storeFullATFLogs then
+    self.full_atf_log_file:write(log_str)
+  end
+end
+
+local function get_script_name(str)
+  local tbl = table.pack(string.match(str, '(.-)([^/]-([^%.]+))$'))
+  local name = tbl[#tbl-1]:gsub('%.'..tbl[#tbl]..'$', '')
+  return name
 end
 
-function Logger:New(path_to_log_file)
-  local logger = { mt = { __index = Logger}}
-  logger.path_to_log_file = path_to_log_file
-  logger.atf_log_file = io.open(path_to_log_file, "w+")
-  setmetatable(logger, logger.mt)
-  return logger
+local function get_log_file_name(timestamp, log_file_type)
+  local dir_name = './' .. Logger.script_file_name
+  local script_name = get_script_name(dir_name)
+  if not timestamp then timestamp = tostring(os.date('%Y%m%d%H%M%S', os.time())) end
+  if (config.reportPath == nil or config.reportPath == '') then
+    config.reportPath = "."
+  end
+  local reportMark = config.reportMark
+  if (reportMark == nil) then reportMark = ''
+  else reportMark = "_" .. reportMark end
+  local curr_log_dir = config.reportPath .. '/' .. log_file_type
+  local curr_log_path = io.catdir(curr_log_dir ..'_'.. timestamp, io.catdir(io.dirname(dir_name)))
+  local full_log_name = io.catfile(curr_log_path, script_name ..'_'..Logger.timestamp .. reportMark)
+  os.execute('mkdir -p "'.. curr_log_path .. '"')
+  return full_log_name
+end
+
+function Logger.init_log(script_name)
+  Logger.script_file_name = script_name
+  local timestamp = tostring(os.date('%Y%m%d%H%M%S', os.time()))
+  local log_file_name = get_log_file_name(timestamp, "ATFLogs") 
+  local atf_log_file_name = log_file_name ..".txt"
+  Logger.atf_log_file = io.open(atf_log_file_name, "r") 
+  if Logger.atf_log_file ~= nil then 
+    io.close(Logger.atf_log_file)
+  end
+  Logger.atf_log_file = io.open(atf_log_file_name, "w+")
+  
+  if config.storeFullATFLogs then
+    local full_atf_log_file_name = log_file_name .. "_full.txt";
+    Logger.full_atf_log_file = io.open(full_atf_log_file_name, "r")
+    if Logger.full_atf_log_file ~= nil then
+      io.close(Logger.full_atf_log_file)      
+    end
+    Logger.full_atf_log_file = io.open(full_atf_log_file_name, "w+")
+  end
+
+  setmetatable(Logger, Logger.mt)
+  is_open = true
+  return Logger
+end
+
+function Logger.LOGTestCaseStart(test_case)
+  if config.excludeReport then return end
+  Logger:StartTestCase(test_case)
+end
+
+function Logger.LOG(tract, message)
+  if config.excludeReport then return end
+  if config.storeFullATFLogs then    
+    Logger[tract](Logger, message)
```

Comment:
@okozlovlux If storeFullATFLogs is true and message from HMI then it should be written twice?